bin/test-external-partner-project.js

```diff
@@ -113,19 +86,32 @@ try {
     debug('Skipping Smoke Test');
   } else {
     debug('Running Smoke Test');
+    try {
+      execExternal(`${useYarn ? 'yarn install' : 'npm install'}`);
+    } catch (e) {
+      debug(e);
+      throw new Error(
+        `Unable to complete install of dependencies for external project ${externalProjectName}`
+      );
+    }
     execExternal(`ember test`, true);
   }
 } catch (e) {
   smokeTestPassed = false;
 }
 
 try {
-  execExternal(`${useYarn ? 'yarn link ember-data' : 'npm link ember-data'}`);
+  debug('Preparing Package To Run Tests Against Commit');
+  insertTarballsToPackageJson(packageJsonLocation);
+  // we must use npm because yarn fails to install
+  // the nested tarballs correctly (just as it fails to pack them correctly)
+  // we rimraf node_modules first because it was previously
+  // installed using yarn's resolution algorithm
+  execExternal(`rm -rf node_modules`);
+  execExternal(`npm install`);
```

Comment:
If we do this does the initial install make a difference? Can we rm rf the ed folders instead of the whole node modules?

Reply:
sadly it has to be the whole of node_modules because otherwise the `npm install` which is required to install from tarball correctly will screw up some of the resolutions of some of the other dependencies